### PR TITLE
Update `updateSuppressionBlock`  for 1.19

### PR DESCRIPTION
**Pull request by fxmorin**
Very basic switch from the old update suppression block to the new one

### CODE REVIEW

File: src/main/java/carpet/mixins/LevelAccessor.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package carpet.mixins;
+
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.redstone.NeighborUpdater;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.gen.Accessor;
+
+@Mixin(Level.class)
+public interface LevelAccessor {
```
Comment (gnembon) [inline src/main/java/carpet/mixins/LevelAccessor.java:9]: I suggest you add that to the existing interface class. I know its called WorldInterface - feel free to rename it.

Comment (fxmorin) [reply src/main/java/carpet/mixins/LevelAccessor.java:9]: omg that's funny, I checked if a levelAccessor already existed. Ya ill rename the current one

File: src/main/java/carpet/CarpetSettings.java [hunk 2]
```diff
@@ -1045,53 +1044,27 @@ public String validate(CommandSourceStack source, ParsedRule<String> currentRule
...
-    public static int getInteger(String s) {
-        try {
-            return Integer.parseInt(s);
-        } catch(NumberFormatException e) {
-            return -1;
-        }
-    }
-
-    private static class updateSuppressionBlockModes extends Validator<String> {
+    private static class updateSuppressionBlockModes extends Validator<Integer> {
```
Comment (altrisi) [inline src/main/java/carpet/CarpetSettings.java:1062]: Should've been caught earlier, class naming.

File: src/main/java/carpet/CarpetSettings.java [hunk 3]
```diff
@@ -1045,53 +1044,27 @@ public String validate(CommandSourceStack source, ParsedRule<String> currentRule
...
-                }
-            }
+        public Integer validate(CommandSourceStack source, ParsedRule<Integer> currentRule, Integer newValue, String string) {
+            if (newValue < -1) newValue = -1;
             return newValue;
         }
-        @Override
-        public String description() {
-            return "Cannot be negative, can be true, false, or # > 0";
-        }
```
Comment (altrisi) [inline src/main/java/carpet/CarpetSettings.java:1094]: Maybe keep some description, so there's some feedback to the user.

File: src/main/java/carpet/CarpetSettings.java [hunk 4]
```diff
@@ -1045,53 +1044,27 @@ public String validate(CommandSourceStack source, ParsedRule<String> currentRule
...
-                    if (parsedInt <= 0) {
-                        updateSuppressionBlockSetting = -1;
-                        return "false";
-                    } else {
-                        updateSuppressionBlockSetting = parsedInt;
-                    }
-                }
-            }
+        public Integer validate(CommandSourceStack source, ParsedRule<Integer> currentRule, Integer newValue, String string) {
+            if (newValue < -1) newValue = -1;
```
Comment (altrisi) [inline src/main/java/carpet/CarpetSettings.java:1065]: Return `null` if invalid, else users will expect their value to be set. `null` tells them it's an invalid value and cancels the operation.

File: src/main/java/carpet/mixins/CollectingNeighborUpdaterAccessor.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import net.minecraft.world.level.redstone.CollectingNeighborUpdater;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.gen.Accessor;
+
+@Mixin(CollectingNeighborUpdater.class)
+public interface CollectingNeighborUpdaterAccessor {
+    @Accessor("count")
+    void setCount(int count);
+    @Accessor("maxChainedNeighborUpdates")
+    int getMaxChainedNeighborUpdates();
```
Comment (altrisi) [inline src/main/java/carpet/mixins/CollectingNeighborUpdaterAccessor.java:12]: Aren't the annotation parameters implicit by the method names? At the end of the day mixin will have to add it to the refmap anyway.

File: src/main/java/carpet/mixins/Level_tickMixin.java [hunk 6]
```diff
@@ -19,14 +20,20 @@
...
+    @Shadow @Final protected NeighborUpdater neighborUpdater;
     CarpetProfiler.ProfilerToken currentSection;
     CarpetProfiler.ProfilerToken entitySection;
 
     Map<EntityType<?>, Entity> precookedMobs = new HashMap<>();
 
+    @Override
+    public NeighborUpdater getNeighborUpdater() {
+        return this.neighborUpdater;
+    }
```
Comment (altrisi) [inline src/main/java/carpet/mixins/Level_tickMixin.java:35]:
```suggestion
    @Override
    @Unique
    public NeighborUpdater getNeighborUpdater() {
        return this.neighborUpdater;
    }
```
I think Unique can/should go there, if so why not, may save us the mixin if mojang adds it some day.

Needs an import.

**Review by altrisi — COMMENTED** [review]
Looks good other than some small things, pending testing from my side.

**Comment by altrisi** [comment]
Update suppression/update order noob trying to test update suppression related PR. Result: ??? (got the log message about suppression, got no idea how to make that stop a portal from being destroyed).

Code makes sense though.

**Review by altrisi — APPROVED** [review]

**Pull request merged**